SEOImages: User Story 1463880, Part 8 (#11249)
dotnet/docs: docs/framework/wpf/advanced/bidirectional-features-in-wpf-overview.md

>  The first document example contains a document that has a number of network share names; for example `\\server1\folder\file.ext`. Whether you have this network link in an Arabic or English document, you always want it to appear in the same way. The following graphic shows the link in an Arabic <xref:System.Windows.FlowDirection.RightToLeft> document.  

Reviewer: ```suggestion
 The first document example contains a document that has a number of network share names; for example `\\server1\folder\file.ext`. Whether you have this network link in an Arabic or English document, you always want it to appear in the same way. The following graphic illustrates using the Span element and shows the link in an Arabic <xref:System.Windows.FlowDirection.RightToLeft> document:
```

  
 ![Graphic that illustrates using the Span element.](./media/bidirectional-features-in-wpf-overview/flow-direction-span-element.png "FlowDocument")  
  
 Because the text is <xref:System.Windows.FlowDirection.RightToLeft>, all special characters, such as the "\\", separate the text in a right to left order. That results in the link not being shown in the correct order, therefore to solve the problem, the text must be embedded to preserve a separate <xref:System.Windows.Documents.Run> flowing <xref:System.Windows.FlowDirection.LeftToRight>. Instead of having a separate <xref:System.Windows.Documents.Run> for each language, a better way to solve the problem is to embed the less frequently used English text into a larger Arabic <xref:System.Windows.Documents.Span>.  
  

>  The following graphic illustrates this.   

Reviewer: ```suggestion
 The following graphic illustrates this by using the Run element embedded in a Span element:
```

>  ![Graphic illustrates the Run element embedded in a Span element.](./media/bidirectional-features-in-wpf-overview/embedded-span-element.png)  

Reviewer: ```suggestion
 ![Graphic that illustrates the Run element embedded in a Span element.](./media/bidirectional-features-in-wpf-overview/embedded-span-element.png)  
```

<a name="FlowDirectionNontext"></a>   
## FlowDirection with Non-text Elements  

>  <xref:System.Windows.FlowDirection> defines not only how text flows in a textual element but also the flow direction of almost every other [!INCLUDE[TLA2#tla_ui](../../../../includes/tla2sharptla-ui-md.md)] element. The following graphic shows a <xref:System.Windows.Controls.ToolBar> that uses a horizontal <xref:System.Windows.Media.LinearGradientBrush> to draw its background.  

Reviewer: ```suggestion

 <xref:System.Windows.FlowDirection> defines not only how text flows in a textual element but also the flow direction of almost every other [!INCLUDE[TLA2#tla_ui](../../../../includes/tla2sharptla-ui-md.md)] element. The following graphic shows a <xref:System.Windows.Controls.ToolBar> that uses a horizontal <xref:System.Windows.Media.LinearGradientBrush> to draw its background with a left to right gradient.  
```

>  The following graphic shows the output of the previous example.  

Reviewer: ```suggestion
 The following graphic shows the output of the previous example with arrows drawn using the `Path` element:
```

>  ![Graphic that illustrates arrows drawn using the path element.](./media/bidirectional-features-in-wpf-overview/arrows-drawn-path-element.png)  

Reviewer: ```suggestion
 ![Graphic that illustrates arrows drawn using the Path element.](./media/bidirectional-features-in-wpf-overview/arrows-drawn-path-element.png)  
```

 [!code-xaml[Numbers#Numbers](~/samples/snippets/csharp/VS_Snippets_Wpf/Numbers/CS/Window1.xaml#numbers)]  
  

>  The following graphic shows the output of the previous sample if you are running in an Arabic version of [!INCLUDE[TLA#tla_mswin](../../../../includes/tlasharptla-mswin-md.md)].  

Reviewer: ```suggestion
 The following graphic shows the output of the previous sample if you're running in an Arabic version of Windows with Arabic and English numbers displayed:
```

 [!code-csharp[NumbersCSharp#NumbersCSharp](~/samples/snippets/csharp/VS_Snippets_Wpf/NumbersCSharp/CSharp/Window1.xaml.cs#numberscsharp)]  
  

>  The following graphic shows what the window looks like for either programming language.  

Reviewer: ```suggestion
 The following graphic shows what the window looks like for either programming language, displaying Arabic numbers:
```
